libtcmu_config.c

```diff
@@ -358,7 +358,8 @@ static void tcmu_parse_options(struct tcmu_config *cfg, char *buf, int len)
 
 int tcmu_load_config(struct tcmu_config *cfg, const char *path)
 {
-	char buf[TCMU_MAX_CFG_FILE_SIZE];
+	char *buf = malloc(TCMU_MAX_CFG_FILE_SIZE);
```

Comment:
Given that the currently shipped config file is 286 bytes large I think a maximum size of 64KiB or something would suffice. As long as it stays on the heap the 2MiB are also okay for the moment. Just when it's on the stack strange things can happen like with valgrind failing to understand what's going on.

In the end the best solution would be to process the config file line wise using a buffer of LINE_MAX bytes. But that change would have been out of the scope of my security oriented review / pull request.